pkg/generator.go

```diff
@@ -560,9 +562,39 @@ func (g *Generator) Generate(ctx context.Context) error {
 		g.printf("}\n")
 	}
 
+	g.generateConstructor()
+
 	return nil
 }
 
+func (g *Generator) generateConstructor() {
+	const constructor = `
+// %[1]sew%[2]s creates a new instance of %[3]s. It also registers a cleanup function to assert the mocks expectations.
```

Comment:
Would it be possible you could use golang templates for this? This is kind of difficult to read and I think the templates would make it nicer.

Reply:
Well I could do that but to be honest I don't like it :D 
```
diff --git a/pkg/generator.go b/pkg/generator.go
index f26c50c..b969a6a 100644
--- a/pkg/generator.go
+++ b/pkg/generator.go
@@ -8,6 +8,7 @@ import (
 	"go/ast"
 	"go/build"
 	"go/types"
+	"html/template"
 	"io"
 	"os"
 	"path/filepath"
@@ -569,9 +570,9 @@ func (g *Generator) Generate(ctx context.Context) error {
 
 func (g *Generator) generateConstructor() {
 	const constructor = `
-// %[1]sew%[2]s creates a new instance of %[3]s. It also registers a cleanup function to assert the mocks expectations.
-func %[1]sew%[2]s(t testing.TB) *%[3]s {
-	mock := &%[3]s{}
+// {{.FuncFirstLetter}}ew{{.MockNameUpperCase}} creates a new instance of {{.MockName}}. It also registers a cleanup function to assert the mocks expectations.
+func {{.FuncFirstLetter}}ew{{.MockNameUpperCase}}(t testing.TB) *{{.MockName}} {
+	mock := &{{.MockName}}{}
 
 	t.Cleanup(func() { mock.AssertExpectations(t) })
 
@@ -592,7 +593,19 @@ func %[1]sew%[2]s(t testing.TB) *%[3]s {
 		break
 	}
 
-	g.printf(constructor, funcFirstLetter, mockNameUpperCase, mockName)
+	err := template.Must(template.New("").Parse(constructor)).Execute(
+		&g.buf,
+		struct {
+			FuncFirstLetter, MockNameUpperCase, MockName string
+		}{
+			FuncFirstLetter:   funcFirstLetter,
+			MockNameUpperCase: mockNameUpperCase,
+			MockName:          mockName,
+		},
+	)
+	if err != nil {
+		panic("failed to write mock constructor: " + err.Error())
+	}
 }
```

Just now I realized I can simplify the "printf" version, what do you think?
```
diff --git a/pkg/generator.go b/pkg/generator.go
index f26c50c..5deb823 100644
--- a/pkg/generator.go
+++ b/pkg/generator.go
@@ -569,9 +569,9 @@ func (g *Generator) Generate(ctx context.Context) error {
 
 func (g *Generator) generateConstructor() {
 	const constructor = `
-// %[1]sew%[2]s creates a new instance of %[3]s. It also registers a cleanup function to assert the mocks expectations.
-func %[1]sew%[2]s(t testing.TB) *%[3]s {
-	mock := &%[3]s{}
+// %[1]s creates a new instance of %[2]s. It also registers a cleanup function to assert the mocks expectations.
+func %[1]s(t testing.TB) *%[2]s {
+	mock := &%[2]s{}
 
 	t.Cleanup(func() { mock.AssertExpectations(t) })
 
@@ -592,7 +592,7 @@ func %[1]sew%[2]s(t testing.TB) *%[3]s {
 		break
 	}
 
-	g.printf(constructor, funcFirstLetter, mockNameUpperCase, mockName)
+	g.printf(constructor, funcFirstLetter+"ew"+mockNameUpperCase, mockName)
 }

```

Reply:
@LandonTClipp Hi, can we please finish this? I'd really love to have the constructors for all my mocks :)